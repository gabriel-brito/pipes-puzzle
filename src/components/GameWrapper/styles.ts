import styled from 'styled-components'

type PipesWrapperTypes = {
  gridColumnSize: number
}

export const Wrapper = styled.section`
  align-items: center;
  background-color: var(--bgContrast);
  border-radius: 8px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 32px auto;
  padding: 16px;
  width: fit-content;
`

export const Information = styled.h2`
  font-size: 14px;
  margin-bottom: 16px;
`

export const PipesWrapper = styled('div')<PipesWrapperTypes>`
  display: grid;
  grid-gap: 8px;
  grid-template-columns: repeat(${(props) => props.gridColumnSize}, 40px);
  margin-bottom: 24px;
  padding: 0 16px;
`

export const Button = styled.button`
  background-color: var(--buttonBg);
  background: linear-gradient(135deg, #ff80bf 0, #9580ff 100%);
  border-radius: 8px;
  border: none;
  color: var(--buttonColor);
  cursor: pointer;
  font-size: 16px;
  font-size: 600;
  font-weight: 900;
  height: 48px;
  margin: 16px auto;
  width: 128px;
}

  &:hover {
    opacity: 0.7;
  }
`
