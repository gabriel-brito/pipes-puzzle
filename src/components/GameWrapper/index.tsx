import { ReactNode } from 'react'
import * as S from './styles'

type GameWrapperTypes = {
  children?: ReactNode
  gridColumnSize: number
  handleStartGame: () => void
  isConnected: boolean
  hasStarted: boolean
}

export default function GameWrapper({
  children,
  gridColumnSize,
  handleStartGame,
  hasStarted,
  isConnected
}: GameWrapperTypes) {
  const message = isConnected
    ? 'The game is connected and ready to play!'
    : 'Loading...'

  return (
    <S.Wrapper data-testid="game-wrapper">
      <S.Information>{message}</S.Information>

      {children && (
        <S.PipesWrapper gridColumnSize={gridColumnSize}>
          {children}
        </S.PipesWrapper>
      )}

      {isConnected && !hasStarted && (
        <S.Button onClick={handleStartGame} aria-label="start-game-button">
          Start Game
        </S.Button>
      )}
    </S.Wrapper>
  )
}
