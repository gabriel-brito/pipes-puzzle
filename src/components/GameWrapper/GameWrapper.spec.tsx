import { render, screen } from '@testing-library/react'
import 'jest-styled-components'

import GameWrapper from '.'

const props = {
  gridColumnSize: 5,
  handleStartGame: jest.fn(),
  hasStarted: false,
  isConnected: true
}

describe('GameWrapper', () => {
  it('should render properly', () => {
    render(<GameWrapper {...props} />)

    expect(
      screen.getByRole('heading', { name: /The game is connected/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /start-game-button/i })
    ).toBeInTheDocument()
    expect(screen.getByTestId('game-wrapper')).toMatchSnapshot()
  })

  it('should render with a different message', () => {
    render(<GameWrapper {...props} isConnected={false} />)

    expect(
      screen.getByRole('heading', { name: /Loading/i })
    ).toBeInTheDocument()
  })

  it('should render with children', () => {
    const Span = () => <span data-testid="test-span">Hello World</span>
    render(<GameWrapper {...props}>{Span()}</GameWrapper>)

    expect(screen.getByTestId('test-span')).toBeInTheDocument()
  })

  it('should render without the button', () => {
    render(<GameWrapper {...props} hasStarted={true} />)

    expect(
      screen.queryByRole('button', { name: /start-game-button/i })
    ).not.toBeInTheDocument()
  })
})
