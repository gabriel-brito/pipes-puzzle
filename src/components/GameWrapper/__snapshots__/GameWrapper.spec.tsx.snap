// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameWrapper should render properly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: var(--bgContrast);
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 32px auto;
  max-width: 1280px;
  padding: 16px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c1 {
  font-size: 14px;
  margin-bottom: 24px;
}

.c2 {
  background-color: var(--buttonBg);
  background: linear-gradient(135deg,#ff80bf 0,#9580ff 100%);
  border-radius: 8px;
  border: none;
  color: var(--buttonColor);
  cursor: pointer;
  font-size: 16px;
  font-size: 600;
  font-weight: 900;
  height: 48px;
  margin-bottom: 16px;
  width: 128px;
}

.c2:hover {
  opacity: 0.7;
}

@media (max-width:800px) {
  .c0 {
    width: 100%;
  }
}

<section
  class="c0"
  data-testid="game-wrapper"
>
  <h2
    class="c1"
  >
    The game is connected and ready to play!
  </h2>
  <button
    aria-label="start-game-button"
    class="c2"
  >
    Start Game
  </button>
</section>
`;
